### PR TITLE
feat(rust, python): supports cast to list

### LINKED ISSUE #10611: exceptions.ArrowErrorException: NotYetImplemented("Casting from ... to LargeList(Field { ... not supported")
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```
df = pl.DataFrame({
	"a": [1,2,3]
})

df = df.with_columns([
	pl.col("a").cast(pl.List(pl.Int64))
])
```

### Issue description

Please implement casting to pl.List from any dtype.

### Expected behavior

Casting to List works for any dtype.

### Installed versions

<details>

```
-------Version info---------
Polars:              0.18.13
Index type:          UInt32
Platform:            Linux-6.1.0-10-amd64-x86_64-with-glibc2.36
Python:              3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
matplotlib:          <not installed>
numpy:               1.25.2
pandas:              <not installed>
pyarrow:             <not installed>
pydantic:            <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by reswqa**
This closes #10611.

### CODE REVIEW

**Comment by reswqa** [comment]
Something is unexpected, let me take a look.

**Comment by reswqa** [comment]
Fixed on upstream in: https://github.com/jorgecarleitao/arrow2/pull/1550

**Comment by reswqa** [comment]
Not entirely certain, but I assume that the failed workflow `Benchmark` is unrelated to this changes. 🤔

**Comment by reswqa** [comment]
@ritchie46 Finally, all checks have passed. 😌

**Review by stinodego — CHANGES_REQUESTED** [review]
Great! Just a minor detail for the test case.

File: py-polars/tests/unit/datatypes/test_list.py [hunk 1]
```diff
@@ -439,6 +439,14 @@ def test_list_recursive_categorical_cast() -> None:
     assert s.to_list() == values
 
 
+def test_non_nested_cast_to_list() -> None:
+    df = pl.DataFrame({"a": [1, 2, 3]})
+
+    df = df.with_columns([pl.col("a").cast(pl.List(pl.Int64))])
+
+    assert df.to_dict(False) == {"a": [[1], [2], [3]]}
```
[the remark this answers is not in the document]
Comment (reswqa) [reply py-polars/tests/unit/datatypes/test_list.py]: Really make sense!

[the remark this answers is not in the document]
Comment (reswqa) [reply py-polars/tests/unit/datatypes/test_list.py]: I have updated this following this way.

**Review by stinodego — APPROVED** [review]
Great, thanks!

**Pull request merged**